server/channels/api4/group.go

```diff
@@ -831,6 +831,10 @@ func getGroupsByTeamCommon(c *Context, r *http.Request) ([]byte, *model.AppError
 		return nil, model.NewAppError("Api4.getGroupsByTeam", "api.ldap_groups.license_error", nil, "", http.StatusForbidden)
 	}
 
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), c.Params.TeamId, model.PermissionManageSystem) {
```

Comment:
Mostly for my education, is there a difference between `SessionHasPermissionToTeam` and `SessionHasPermissionTo` if the permission we're checking in the `PermissionManageSystem` ? I'd expect there's not a difference.

Reply:
As this is a public endpoint, it should be able to be used outside of the System Console too. Isn't the `manage_system` too strict? My concern is that this could break customers who use the API to get MM data out. 

I'd at least lessen it to Team Admin permissions but I'd also be aligned with checking team membership.

Reply:
> Mostly for my education, is there a difference between SessionHasPermissionToTeam and SessionHasPermissionTo if the permission we're checking in the PermissionManageSystem?

I don't think there is really, as it should not be part of the team membership.

> I'd at least lessen it to Team Admin permissions but I'd also be aligned with checking team membership.

Discussed offline, decided to go for `model.PermissionListTeamChannels`